### ISSUE TITLE
Java code not Compiling

### ISSUE THREAD

**Issue by ArsalanWahid** [atom-community/atom-script#481]
Hi i just installed script for atom 1.0 and tried running some code first for python which worked perfectly ,but when I made some code for java I'am face this error Posted in the snap.
The file Extension  is right and so is the language set to java.

Hope for a response soon thanks 
![snap1](https://cloud.githubusercontent.com/assets/13076610/8398744/f9a22ed0-1e14-11e5-9c6c-87e08bd39cab.JPG)

**Comment by rgbkrk** [comment]
The problem there is that the contributed Java grammar uses [bash as a workaround to let you run a single Java file](https://github.com/rgbkrk/atom-script/blob/master/lib/grammars.coffee#L131-L137):

```
  Java:
    "File Based":
      command: "bash"
      args: (context) ->
        className = context.filename.replace /\.java$/, ""
        args = ['-c', "javac -d /tmp #{context.filepath} && java -cp /tmp #{className}"]
        return args
```

As mentioned in #441, this isn't cross platform by any means.

/cc @pardom

**Comment by Oalecj** [comment]
@rgbkrk Or you could install Cygwin - I did that and never encountered any errors (other than #480)

**Comment by ArsalanWahid** [comment]
Thank You @rgbkrk

**Comment by XSStudio** [comment]
You must be running in Windows (same as me~). Besides installing Cygwin, you can modify "grammars.coffee" in "%userprofile%\ .atom\packages\script\lib" to the following:
```coffeescript
Java:
  "File Based":
    command: "cmd"
    args: (context) ->
      className = context.filename.replace /\.java$/, ""
      args = ["/c javac -Xlint #{context.filename} && java #{className}"]
      return args
```

**Comment by Javacym** [comment]
javac: file not found: Test.java
Usage: javac <options> <source files>
use -help for a list of possible options
[Finished in 0.68s]  
why?

**Comment by XSStudio** [comment]
@Javacym Check your path: does it contain characters besides your system locale?

**Comment by rgbkrk** [comment]
@XSStudio I'd welcome a patch that conditionally executed that code based on the operating system. There are a few other languages that change what their grammar definition is based on what platform they're on.

**Comment by XSStudio** [comment]
@rgbkrk Of course, here you go~ (I just didn't realize script already has platform detection function...)

**Comment by robinhur** [comment]
In Ubuntu, Java Scanner Object doesn't work!!!
When I run code included Scanner with java.util.Scanner imported, Code cannot be run!!
Plz Help me!! I just want my scanner!!

**Comment by amankriet** [comment]
@XSStudio Thanks a lot for that modification code of yours for script. I just added "start cmd /k" before classname in return statement and now my java programs are set to take inputs!

Could you please tell me how could i set cmd to exit by itself after execution of the program just as it happens in C... Thanks in advance! :)

**Comment by aminya** [comment]
Should be fixed in #2470. Let me know if it is not.
